app/resources/miner.py

```diff
@@ -50,27 +42,20 @@ def set_wallet(data: dict, user: str) -> dict:
     if not exists_wallet(wallet_uuid):
         return wallet_not_found
 
-    mined_coins: int = miner.update_miner()
-    if mined_coins > 0:
-        m.contact_microservice("currency", ["put"], {
-            "destination_uuid": miner.wallet,
-            "amount": mined_coins,
-            "create_transaction": False
-        })
+    update_miner(miner)
 
-    miner.wallet: str = wallet_uuid
+    miner.wallet = wallet_uuid
     wrapper.session.commit()
 
     return miner.serialize
 
 
-@m.user_endpoint(path=["miner", "power"], requires={
-    "service_uuid": UUID(),
-    "power": Integer(minimum=0, maximum=100)
-})
+@m.user_endpoint(path=["miner", "power"], requires={"service_uuid": UUID(), "power": Float(minimum=0, maximum=1)})
 def set_power(data: dict, user: str) -> dict:
     service_uuid: str = data["service_uuid"]
     power: int = data["power"]
+    if power < 0 or power > 1:
+        return {"error": "power_invalid_value"}
```

Comment:
this is not possible because of the requirements, so delete this

Reply:
@Defelo I already changes the requirements ...